### DISCUSSION TITLE
Mapping classes in transfer learning

### DISCUSSION THREAD

**Discussion by eskildbr** [ultralytics/yolov5#5837] [Q&A]
Hi, 

So me and a friend are training our own dataset using yolov5, transfer learning, and pre-trained on yolov5 dataset. We have a question about how the code maps the pre-trained dataset using x-amount of classes to our own classes. We have 3 classes we train with, how are these classes mapped from the pre-trained classes?

Also, I have a question about the mAP score we achieved. When done with training we achieve a mAP of 0.90-ish. This is quite high, at least compared to yolov5 or coco where their mAP is around 0.5. Could an explanation be that we have way fewer classes that we want to classify? We have quite a few validation images also, maybe that could be a reason.

Thanks! 😄

**Comment by glenn-jocher** [comment]
@eskildbr classes are not 'mapped' when transfer learning. YOLOv5 automatically restructures the architecture to accommodate the dataset class count, no action is required on your part.

For mAP improvements and dataset guidelines see **Tips for Best Training Results** tutorial:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Reply by eskildbr** [reply]
Ah okey I see, thanks for the quick reply!

I would for sure recommend yolov5 for anyone trying out object detection and training with own dataset, really easy to use.
